src/types.ts

```diff
@@ -30,28 +30,41 @@ export type BabelConfig = _babel.TransformOptions
 export interface TsJestGlobalOptions {
   /**
    * Compiler options. It can be:
-   *     - `true` (or `undefined`, it's the default): use default tsconfig file
-   *     - `false`: do NOT use default config file
-   *     - `path/to/tsconfig.json`: path to a specific tsconfig file (<rootDir> can be used)
-   *     - `{...}`: an object with inline compiler options
+   * - `false`: do NOT use default config file
+   * - `path/to/tsconfig.json`: path to a specific tsconfig file (<rootDir> can be used)
+   * - `{...}`: an object with inline compiler options
+   * @default undefined uses the default tsconfig file
+   * @alias tsconfig
    */
-  tsConfig?: boolean | string | _ts.CompilerOptions
+  tsConfig?: false | string | _ts.CompilerOptions
+
+  /**
+   * Compiler options. It can be:
+   * - `false`: do NOT use default config file
+   * - `path/to/tsconfig.json`: path to a specific tsconfig file (<rootDir> can be used)
+   * - `{...}`: an object with inline compiler options
+   * @default undefined uses the default tsconfig file
+   * @alias tsConfig
+   */
+  tsconfig?: false | string | _ts.CompilerOptions
 
   /**
    * packageJson. It can be:
-   *     - `true` (or `undefined`, it's the default): use default package.json file
-   *     - `path/to/package.json`: path to a specific package.json file (<rootDir> can be used)
-   *     - `{...}`: contents of a package.json
+   * - `path/to/package.json`: path to a specific package.json file (<rootDir> can be used)
+   * - `{...}`: contents of a package.json
+   * @default undefined uses the default package.json file
    */
-  packageJson?: boolean | string | object
+  packageJson?: false | string | object
```

Comment:
Replacing `boolean` type with `false` reminds the user that the default value is `true`; thus, it would be redundant to supply `true` in this case.

Reply:
This would be misleading. A type of `false` implies that it will always be `false` and not that the default value will be `true`

Reply:
1. An optional property with a union type of `false | string | object` does not at all imply that it will _always_ be false.
1. The default value is not `true`, but `undefined`, according to [this line](https://github.com/kulshekhar/ts-jest/blob/7f85bab6fc13cdfaf6eef711e094776f98bf150b/src/config/config-set.ts#L245).
1. This is an option. The default value is `undefined`. You may override that value with `false | string | object`. A value of `true` would be superfluous and redundant; thus, misleading.
1. The proper way to indicate default values is with the `@default` tag.

Reply:
My comment above was only in the context of the `false` part of the sum type defined. The other two (string/object) are not relevant to this discussion.

Even if a value of `true` would be redundant, the fact that it can be set means that the type should be boolean and not false. Even though it's redundant, adding that makes it clear in the user's setup what is happening. If it's true by virtue of being absent, it's not always clear - the user will have to know or will have to read the documentation to understand the behavior.

I agree with 4 and that's also part of the reason I'm in favor of leaving the type as boolean instead of false.

Reply:
I'm not interested in fighting this, so I restored the Booleans.